src/include/ipc4/base_fw.h

```diff
@@ -485,7 +485,7 @@ struct ipc4_astate_table {
 struct ipc4_dma_control {
 	uint32_t node_id;
 	uint32_t config_length;
-	uint32_t config_data[1];
+	uint32_t config_data[0];
```

Comment:
I think this change affects the `basefw_dma_control()` function. The following line from from base_fw.c:
`591        data_size = data_offset - (sizeof(struct ipc4_dma_control) - sizeof(uint32_t));`
should be modified e.g. into this:
`591        data_size = data_offset - sizeof(struct ipc4_dma_control);`
to stay aligned to your change.

Reply:
Fixed. Thanks